prov/tcp/src/tcpx_ep.c

```diff
@@ -75,7 +77,9 @@ static ssize_t tcpx_recvmsg(struct fid_ep *ep, const struct fi_msg *msg,
 	recv_entry->context = msg->context;
 	recv_entry->done_len = 0;
 
+	fastlock_acquire(&tcpx_ep->queue_lock);
 	dlist_insert_tail(&recv_entry->entry, &tcpx_ep->rx_queue);
+	fastlock_release(&tcpx_ep->queue_lock);
```

Comment:
This looks like a real problem separate from progress.  A fix for synchronization issues should go into its own patch.

Reply:
Created a separate commit. Thanks